pkg/config/types/v1alpha1/generate/generate.go

```diff
@@ -131,98 +169,76 @@ type TrustdInfo struct {
 	Token string
 }
 
-func generateCertificateKey() (string, error) {
-	key, err := randBytes(32)
-	if err != nil {
-		return "", err
+func NewEtcdCA() (ca *x509.CertificateAuthority, err error) {
+	opts := []x509.Option{
+		x509.RSA(true),
+		x509.Organization("etcd"),
+		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
 	}
 
-	hashedKey := sha256.Sum256([]byte(key))
-	encoded := hex.EncodeToString(hashedKey[:])
-
-	return encoded, nil
+	return x509.NewSelfSignedCertificateAuthority(opts...)
 }
 
-// randBytes returns a random string consisting of the characters in
-// validBootstrapTokenChars, with the length customized by the parameter
-func randBytes(length int) (string, error) {
-	// validBootstrapTokenChars defines the characters a bootstrap token can consist of
-	const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
-
-	// len("0123456789abcdefghijklmnopqrstuvwxyz") = 36 which doesn't evenly divide
-	// the possible values of a byte: 256 mod 36 = 4. Discard any random bytes we
-	// read that are >= 252 so the bytes we evenly divide the character set.
-	const maxByteValue = 252
-
-	var (
-		b     byte
-		err   error
-		token = make([]byte, length)
-	)
-
-	reader := bufio.NewReaderSize(rand.Reader, length*2)
-
-	for i := range token {
-		for {
-			if b, err = reader.ReadByte(); err != nil {
-				return "", err
-			}
+func NewKubernetesCA() (ca *x509.CertificateAuthority, err error) {
+	opts := []x509.Option{
+		x509.RSA(true),
+		x509.Organization("kubernetes"),
+		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
+	}
 
-			if b < maxByteValue {
-				break
-			}
-		}
+	return x509.NewSelfSignedCertificateAuthority(opts...)
+}
 
-		token[i] = validBootstrapTokenChars[int(b)%len(validBootstrapTokenChars)]
+func NewTalosCA() (ca *x509.CertificateAuthority, err error) {
+	opts := []x509.Option{
+		x509.RSA(false),
+		x509.Organization("talos"),
+		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
 	}
 
-	return string(token), err
+	return x509.NewSelfSignedCertificateAuthority(opts...)
 }
 
-// genToken will generate a token of the format abc.123 (like kubeadm/trustd), where the length of the first string (before the dot)
```

Comment:
It is better to place unexpected functions at the bottom of the file.

Reply:
I hope these are `unexported` :)

Reply:
lol yes